android/app/src/main/java/org/fossasia/openevent/core/feedback/FeedbackFragmentViewModel.java

```diff
@@ -33,6 +40,37 @@ public FeedbackFragmentViewModel() {
         return feedbackLiveRealmData;
     }
 
+    public LiveData<Integer> postFeedback(float rating, String comments) {
+        if (postFeedbackResponse == null) {
+            postFeedbackResponse = new MutableLiveData<>();
+        }
+        User user = realmRepo.getUserSync();
+        Event event = realmRepo.getEventSync();
+        Realm realm = Realm.getDefaultInstance();
+        Event eventCopy = realm.copyFromRealm(event, 0);
+        User userCopy = realm.copyFromRealm(user,0);
```

Comment:
Getting this in sync is doing it on UI thread, which is not good

Reply:
This is synchronous as well

Reply:
```
try(Realm realm = Realm.getDefaultInstance()) {
    realm.executeTransaction(realm1 -> { 
        final User user = realm1.where(User.class).findFirst();     
    });
```
is this okay?

Reply:
This is all synchronous code. In async code, you get the user in form of a callback or observable

Reply:
I am using following to fetch `user` and `event`
```
RealmResults<User> user = realm.where(User.class).findAllAsync();
        realm.where(Event.class).findAllAsync().asFlowable().subscribe(realmObject -> {
            Timber.d("user id is "+realmObject.first().getId());
                }
        );

        user.asFlowable().subscribe(users -> {
            Timber.d("event id is "+users.first().getId());

        });
```
since i need both event and user data for posting feedback,
also i would have to convert these into unmanaged objects before passing them to API call
```
 Event eventCopy = realm.copyFromRealm(eventcpy,0);
        User userCopy = realm.copyFromRealm(usercpy,0);
    
        if (user == null || event == null) {
           postFeedbackResponse.setValue(ON_ERROR);
        } else {
            compositeDisposable.add(APIClient.getOpenEventAPI().postFeedback(Feedback.builder().event(eventCopy)
                    .rating(Float.toString(rating)).comment(comments).user(userCopy).build())
                    .subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())
                    .subscribe(feedback -> {
                                postFeedbackResponse.setValue(ON_SUCCESS);
                                Timber.d("Posted successfully");
                            },
                            throwable -> {
                                postFeedbackResponse.setValue(ON_ERROR);
                                Timber.d("Error posting feedback " + throwable.toString());
                            }));
        }
```
 Problems I am facing
1- There are two callbacks, when to start manged object to unmanage coversion and call the API what would be the logic since I also cannot update a global variable from inside lambda expression
2- This is returning empty response
```
 realm.where(Event.class).findAllAsync().asFlowable().subscribe(realmObject -> {
            Timber.d("harsimar id is "+realmObject.first().getId());
                }
        );
```
@iamareebjamal @nikit19 @dreadpool2 please help me out on this one

Reply:
You can chain RxJava observable, but ok you can use synchronous query here

Reply:
No changes are needed then I believe?